### DISCUSSION TITLE
Integrate with Apache StreamPark

### DISCUSSION THREAD

**Discussion by zhoujinsong** [apache/amoro#2026] [Ideas]
Hi, community

Last Friday (2023-09-22), an online meeting was held between the Amoro community and the [Apache StreamPark](https://streampark.apache.org/) community to initiate integration and cooperation between the two projects.

After having a sufficient understanding of both projects, the two communities have developed preliminary plans for the subsequent integration:

1. Manually start Amoro Flink Optimizer in StreamPark to connect the two systems.This step does not require any additional development work.
2. Implement `StramParkOptimizerContainer` in Amoro project to automatically start/stop Amoro Optimizer in StreamPark.
3. Integrate Amoro's table management functionality into StreamPark to facilitate task writing for StreamPark users.This is still in the ideation stage, and further discussion is needed on how to design and implement it.

We will start planning and implementing the integration of the two projects step by step. Anyone who is interested is welcome to participate in the integration work. If you have any ideas about the integration of the two projects, please let us know.

**Comment by wolfboys** [comment]
I am very excited about the collaboration with the community. StreamPark can submit jobs that manage Amoro Flink Optimizer types, and the table/catalog management functionality in Amoro is what Streampark needs. Looking forward to a win-win collaboration.
